flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1449,6 +1449,7 @@ abstract class CodeGenerator(
     val resultCode =
       s"""
         |long $resultTerm = $contextTerm.timerService().currentProcessingTime();
+        |$resultTerm = $resultTerm + java.util.TimeZone.getDefault().getOffset($resultTerm);
```

Comment:
Thanks for your pr. We can't add the offset here. The long type time has already take timezone into consideration. The result is wrong if we output the long value directly.